### PR TITLE
Consolidate Heading Components into a Single Typography Component

### PR DESCRIPTION
**Pull request by AlSirang**
## Consolidate Heading Components into a Single `Text` Component

This pull request consolidates individual heading components (`H1`, `H2`, `H3`, `H4`, `H5`, and `H6`) into a single, flexible `Text` component. This approach streamlines the implementation of headings while maintaining semantic HTML and accessibility.

**Features:**

- **Single `Text` Component:** A unified component that allows users to specify the desired heading level (from `h1` to `h6` and `p`) using props, while applying the appropriate styles for each level.
- **Variant Support:** Utilizes class-variance-authority to handle styling variants, ensuring consistent typography across different heading levels.
- **Flexible Props:** Supports `className`, `component`, and additional HTML attributes, allowing for extensive customization and integration.
- **Improved Documentation:** Updated documentation includes usage examples and guidelines for the new consolidated component, making it easier for developers to adopt.

**Usage Example:**

```jsx
import { Text } from "./Text";

const ExampleComponent = () => {
  return (
    <div>
      <Text as="h1">This is an H1 Heading</Text>
      <Text as="h2">This is an H2 Heading</Text>
      <Text as="h3">This is an H3 Heading</Text>
      <Text as="h4">This is an H4 Heading</Text>
      <Text as="h5">This is an H5 Heading</Text>
      <Text as="h6">This is an H6 Heading</Text>
      <Text>This is a paragraph Element</Text>
    </div>
  );
};
```

### CODE REVIEW

**Comment by ariflogs** [comment]
Good idea. thanks for the work. 
But don't you think naming the component something like `Heading` would be better?

**Comment by AlSirang** [comment]
Yes, we could have used a component name like `Heading`, but in the future, we can add a paragraph to the `Typography` component to cover all text tags.

**Comment by ariflogs** [comment]
like every typography types(heading, p, quotes, code...) all in the same component? 
that might start to get a little confusing for the users.

**Comment by ariflogs** [comment]
btw are you in our Discord server?

**Comment by AlSirang** [comment]
> like every typography types(heading, p, quotes, code...) all in the same component? 
> that might start to get a little confusing for the users.

No, just `h1` to `h6` and `p` tag. Not all of them. Just like the MUI Typography component.

**Comment by AlSirang** [comment]
> btw are you in our Discord server?
No, not yet.

**Comment by ariflogs** [comment]
makes sense. I started a poll on the community server. let's see if others have any opinion on this.

File: packages/ui/Text/Text.tsx
Comment (ariflogs) [inline packages/ui/Text/Text.tsx:1]: Looks like we don't need moth `variant` and `tag`, let's combine both and make it `as`

File: packages/ui/Text/Text.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import type { ElementType, HTMLAttributes } from "react";
+import { type VariantProps, cva } from "class-variance-authority";
+import { cn } from "@/lib/utils";
+
+const textVariants = cva("font-head", {
+  variants: {
+    variant: {
```
Comment (ariflogs) [inline packages/ui/Text/Text.tsx:7]: variants: {
  as: {
    p: ""
    }
....

File: packages/ui/Text/Text.tsx
Comment (AlSirang) [reply packages/ui/Text/Text.tsx:1]: The reason for both the `tag` and `variants` is to allow users to apply sizes and styles from other elements to the specified `tag`. If we remove the `variants`, we will lose that feature.

File: packages/ui/Text/Text.tsx (hunk 1 above)
Comment (AlSirang) [reply packages/ui/Text/Text.tsx:7]: You mean I should update the `variant` prop to `as`?

**Review by ariflogs — APPROVED** [review]
Thanks for the great work 🚀

**Pull request merged**